tex/polyglossia.sty

```diff
@@ -1244,33 +1234,41 @@
 }
 
 \newcommand\xpg@ifdefined[3]{%
-    \ifluatex
-        \xpg@set@language@luatex@iv{#1}%
-    \fi
+    \xpg@set@hyphenation@patterns{#1}%
     \polyglossia@check@ifdefined:NTF{#1}{#2}{#3}
 }%
 
-\newcommand\xpg@set@language@luatex@iv[1]{%
+\newcommand*\xpg@set@bbl@hyphendata[1]{%
+    % here we use lu@texhyphen@loaded@\the\language, the same as in babel
+    \ifcsdef{bbl@hyphendata@#1}{}{%
+        \global\@namedef{bbl@hyphendata@\the\language}{}%
+    }
+}
+
+\newcommand\xpg@set@hyphenation@patterns[1]{%
     \ifluatex
         \polyglossia@check@ifdefined:NF{#1}{
             \expandafter\chardef\csname l@#1\endcsname=\directlua{tex.sprint(polyglossia.newloader('#1'))}%
         }
-        \language=\csname l@#1\endcsname
     \fi
+    \polyglossia@check@ifdefined:NF{#1}{
+        \language=\csname l@#1\endcsname
+    }
 }
 
+% FIXME: Remove after gloss-latin (last user) has been rewritten)
 \newcommand\xpg@set@language@luatex@ii[1]{%
    \ifluatex %
-     \xpg@set@language@luatex@iii{#1}{\expandafter\the\csname l@#1\endcsname}
+     \xpg@set@bbl@hyphendata{\expandafter\the\csname l@#1\endcsname}
    \fi % 
 }
 
 
 \def\@select@language#1{
-  % hook for compatibility with biblatex
-  \select@language{#1}
-  \ifluatex %
-      \xpg@set@language@luatex@iii{#1}{\the\language}
+   % hook for compatibility with biblatex
+   \select@language{#1}
+   \ifluatex %
```

Comment:
Push this \iflualatex in \xpg@set@bbl@hyphendata and execute inconditionnaly \xpg@set@bbl@hyphendata